SU2_DEF/src/drivers/CDeformationDriver.cpp

```diff
@@ -0,0 +1,804 @@
+
+    if (rank == MASTER_NODE) cout << endl << "----------------------- Write deformed grid files -----------------------" << endl;
+
+    for (iZone = 0; iZone < nZone; iZone++){
+
+      /*--- Compute Mesh Quality if requested. Necessary geometry preprocessing re-done beforehand. ---*/
+
+      if (config_container[iZone]->GetWrt_MeshQuality() && !driver_config->GetStructuralProblem()) {
+
+        if (rank == MASTER_NODE) cout << "Recompute geometry properties necessary to evaluate mesh quality statistics.\n";
+
+        geometry_container[iZone]->SetPoint_Connectivity();
+        geometry_container[iZone]->SetBoundVolume();
+        geometry_container[iZone]->SetEdges();
+        geometry_container[iZone]->SetVertex(config_container[iZone]);
+        geometry_container[iZone]->SetControlVolume(config_container[iZone], ALLOCATE);
+        geometry_container[iZone]->SetBoundControlVolume(config_container[iZone], ALLOCATE);
+
+        if (rank == MASTER_NODE) cout << "Computing mesh quality statistics for the dual control volumes.\n";
+        geometry_container[iZone]->ComputeMeshQualityStatistics(config_container[iZone]);
+      }// Mesh Quality Output
+
+      /*--- Load the data --- */
+
+      output_container[iZone]->Load_Data(geometry_container[iZone], config_container[iZone], nullptr);
+
+      output_container[iZone]->WriteToFile(config_container[iZone], geometry_container[iZone], MESH, driver_config->GetMesh_Out_FileName());
+
+      /*--- Set the file names for the visualization files ---*/
+
+      output_container[iZone]->SetVolume_Filename("volume_deformed");
+      output_container[iZone]->SetSurface_Filename("surface_deformed");
+
+      for (unsigned short iFile = 0; iFile < config_container[iZone]->GetnVolumeOutputFiles(); iFile++){
+        auto FileFormat = config_container[iZone]->GetVolumeOutputFiles();
+        if (FileFormat[iFile] != RESTART_ASCII && FileFormat[iFile] != RESTART_BINARY)
+          output_container[iZone]->WriteToFile(config_container[iZone], geometry_container[iZone], FileFormat[iFile]);
+      }
+    }
+
+    if (!driver_config->GetDeform_Mesh()) {
+      if ((config_container[ZONE_0]->GetDesign_Variable(0) != NO_DEFORMATION) &&
+          (config_container[ZONE_0]->GetDesign_Variable(0) != SCALE_GRID)     &&
+          (config_container[ZONE_0]->GetDesign_Variable(0) != TRANSLATE_GRID) &&
+          (config_container[ZONE_0]->GetDesign_Variable(0) != ROTATE_GRID)) {
+
+          /*--- Write the the free-form deformation boxes after deformation. ---*/
+
+          if (rank == MASTER_NODE) cout << "Adding any FFD information to the SU2 file." << endl;
+
+          surface_movement[ZONE_0]->WriteFFDInfo(surface_movement, geometry_container, config_container);
+
+    }
+  }
+}
+
+void CDeformationDriver::Postprocessing() {
+
+    if (rank == MASTER_NODE)
+      cout << endl <<"------------------------- Solver Postprocessing -------------------------" << endl;
+
+    delete driver_config;
```

Comment:
make this the destructor and remove the Postprocessing, it should clear the LGTM warning you are getting at the moment.